### PR TITLE
update dev dependency versions

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

Keeping dev tools up-to-date

### 🤓 What?

- update dev dependency versions
- add `bandit` for code security check

### 🧪 Tested?

bandit is working for pre-commit check. Will see if the CI works

### CODE REVIEW

File: tests/common/test_extractor.py [hunk 1]
```diff
@@ -28,7 +27,6 @@ async def extract(self, config: BaseConfig) -> List[MetadataChangeEvent]:
         return self._dummy_events
 
 
-@pytest.mark.asyncio
```
Comment (mars-lan) [inline tests/common/test_extractor.py:31]: No longer needed?

Comment (alyiwang) [reply tests/common/test_extractor.py:31]: right, no async execution in this test

Comment (mars-lan) [reply tests/common/test_extractor.py:31]: But it ultimately called the async `extract`? Will that be an issue?

Comment (alyiwang) [reply tests/common/test_extractor.py:31]: It calls the `run` which is not async, also underlying is a dummy function that's not async either.

Comment (alyiwang) [reply tests/common/test_extractor.py:31]: This is a warning thrown by the upgraded `pytest-asyncio`

Comment (mars-lan) [reply tests/common/test_extractor.py:31]: Got it.

**Review by mars-lan — APPROVED** [review]

**Pull request merged**